JCL-67: CI for live testing

timea-solid opened the pull request:
This is based on 2 environments set up on GitHub secrets: ESS PodSpaces and ESS DevNext. Each of them defines:
* INRUPT_TEST_WEBID
* INRUPT_TEST_CLIENT_ID
* INRUPT_TEST_CLIENT_SECRET
* INRUPT_TEST_PUBLIC_RESOURCE_PATH

These are then read in the tests through the CI.
The CI is taken from a [Javascript project](https://github.com/inrupt/solid-client-js/blob/main/.github/workflows/e2e-browser.yml) (builds on top of their learnings). For example, DevNext tests are not gating on purpose.

timea-solid committed:
connecting gitHub secrets with CI for integration tests

timea-solid committed:
Merge branch 'main' into JCL-67-ci

timea-solid committed:
moved env use in verify step of CI

timea-solid committed:
java version also for devNext

timea-solid committed:
only one java version for devNext

timea-solid committed:
rearange steps in ci

NSeydoux reviewed (changes requested):
The `experimental` flag on the matrix should be used in order to determine whether the job failing should fail the workflow or not: https://github.com/inrupt/solid-client-access-grants-js/blob/57f050215d6f5c3d7459c4b49e2d59bacfaec24b/.github/workflows/e2e-node.yml#L20. In other words, that's an ad-hoc configuration we use to set a standard value, rather than being a standard value in the first place.

timea-solid committed:
added experimental keyword to matrix

timea-solid committed:
added experimental keyword to matrix

timea-solid committed:
improved CI

timea-solid committed:
try to jumpstart dependabot for my branch

langsamu committed:
Distinguish between dependabot and normal PRs

Duplicate the build step and exclude integration modules when run by dependabot

timea-solid committed:
remove target branch from dependabot

timea-solid committed:
Merge branch 'JCL-67-ci' of https://github.com/inrupt/solid-client-java into JCL-67-ci

timea-solid committed:
rename of experimental value

langsamu committed:
Restructure dependabot differentiation

Everything except integration runs always.
Integration runs only if not dependabot

timea-solid committed:
revert experiemental to boolean values

NSeydoux reviewed (approved):
Looks good, although maybe a comment on the Maven flags used to distinguish between unit tests and integration ones would be helpful? And same, can you make sure that there is either a comment or a commit message that explains why dependabot PRs skip integration tests, otherwise that'll definitely confuse future us.
